libs/fetcher.client.js

```diff
@@ -49,456 +37,278 @@ function parseResponse(response) {
-            return this;
-        }
-
-        if (this.config.wait <= 0) {
-            // process immediately
-            this._cb(item);
-            this._items = [];
-            return this;
-        }
-
-        var self = this;
-        this._items.push(item);
-
-        // setup timer
-        if (!this._timer) {
-            this._timer = setTimeout(function sweepInterval() {
-                var items = self._items;
-                self._items = [];
-                clearTimeout(self._timer);
-                self._timer = null;
-                items = self._sweep(items);
-                lodash.forEach(items, function eachItem(item) {
-                    self._cb(item);
-                });
-            }, this.config.wait);
-        }
-        return this;
-    }
+Request.prototype.body = function (body) {
+    this._body = body;
+    return this;
 };
-
-    /**
-     * Requests that are initiated within a time window are batched and sent to xhr endpoint.
-     * The received responses are split and routed back to the callback function assigned by initiator
-     * of each request.
-     *
-     * All requests go out from this store is via HTTP POST. Typical structure of the context param is:
-     * <pre>
-     * {
-     *   config: {
-     *     uri : '/api'
-     *   },
-     *   context: {
-     *     _csrf : '5YFuDK6R',
-     *     lang : 'en-US',
-     *     ...
-     *   }
-     * }
-     * </pre>
-     *
-     * @class FetcherClient
-     * @param {object} options configuration options for Fetcher
-     * @param {string} [options.xhrPath="/api"] The path for XHR requests
-     * @param {number} [options.xhrTimout=3000] Timeout in milliseconds for all XHR requests
-     * @param {number} [options.batchWindow=20] Number of milliseconds to wait to batch requests
-     * @param {Boolean} [options.corsPath] Base CORS path in case CORS is enabled
-     * @param {Object} [options.context] The context object that is propagated to all outgoing
-     *      requests as query params.  It can contain current-session/context data that should
-     *      persist to all requests.
-     */
-
-    function Fetcher (options) {
-        this.xhrPath = options.xhrPath || DEFAULT_XHR_PATH;
-        this.xhrTimeout = options.xhrTimeout || DEFAULT_XHR_TIMEOUT;
-        this.corsPath = options.corsPath;
-        this.batchWindow = options.batchWindow || DEFAULT_BATCH_WINDOW;
-        this.context = options.context || {};
+/**
+ * @method clientConfig
+ * @memberof Request
+ * @param {Object} config config for this fetcher request
+ */
+Request.prototype.clientConfig = function (config) {
+    this._clientConfig = config;
+    return this;
+};
+/**
+ * Execute this fetcher request and call callback.
+ * @method end
+ * @memberof Request
+ * @param {Fetcher~fetcherCallback} callback callback invoked when fetcher/service is complete.
+ */
+Request.prototype.end = function (callback) {
+    var clientConfig = this._clientConfig || {};
```

Comment:
@Vijar meaning you are setting a default here, but that is already done in the constructor. are you just being save in case the user sets clientConfig incorrectly?